src/dygraph.js

```diff
@@ -3167,6 +3168,7 @@ Dygraph.prototype.updateOptions = function(input_attrs, block_redraw) {
 
   this.attributes_.reparseSeries();
 
+  if (prevNumAxes < this.attributes_.numAxes()) this.plotter_.clear();
```

Comment:
Adding the condition to requiresNewPoints does not solve the problem.

prevNumAxes != this.attributes_.numAxes() will work, however if you are changing from 2 axes to 1 the graph will overwrite the second axis so calling clear() is not necessary.